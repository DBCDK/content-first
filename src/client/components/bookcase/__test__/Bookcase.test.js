import React from 'react';
import renderer from 'react-test-renderer';

import {Bookcase} from '../Bookcase.component';

jest.mock(
  '../../general/CheckmarkConnected.component',
  () => 'CheckmarkConnected'
);
jest.mock('../../pulse/Pulse.component', () => 'Pulse');
jest.mock('../../rollover/RollOver.component', () => 'RollOver');

const bookcaseStateOBJ = {
  books: [
    {
      pid: '870970-basis:52038014',
      position: {x: 26, y: 46},
      description: 'lorem ipsum. . .'
    },
    {
      pid: '870970-basis:23211629',
      position: {x: 36.5, y: 46},
      description: 'lorem ipsum. . .'
    }
  ]
};

const workStateOBJ = {
  pid: '870970-basis:52530423',
  creator: 'Nicole Boyle RÃ¸dtnes',
  title: 'test',
  cover: 'img/cover.png',
  description: 'lorem ipsum. . .'
};

describe('Bookcase', () => {
  it('renders initial component', () => {
    const tree = renderer
      .create(
        <Bookcase bookcaseState={bookcaseStateOBJ} booksState={workStateOBJ} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
