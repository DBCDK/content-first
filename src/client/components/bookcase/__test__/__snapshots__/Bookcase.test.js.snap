// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookcase renders initial component 1`] = `
<section
  className="row "
  onClick={undefined}
>
  <img
    alt="bogreol"
    src="img/bookcase/BS-bogreol.png"
  />
  <div
    className="row"
  >
    <div
      className="col-xs-4 celeb"
    >
      <img
        alt="luk"
        className="carousel-close"
        onClick={[Function]}
        src="/static/media/Kryds.e69a54ef.svg"
      />
      <div
        className="col-xs-12 celeb-top"
      >
        <div
          className="col-xs-12 celeb-img"
        >
          <img
            alt="bs"
            src="img/bookcase/BS3.png"
          />
        </div>
        <div
          className="col-xs-12 celeb-title"
        >
          <h1>
            Bjarne Slot Christiansen
          </h1>
        </div>
        <div
          className="col-xs-12 celeb-description"
        >
          <p>
            er en dansk forhenværende elitesoldat i Jægerkorpset og fordragsholder. Han er især kendt for programserien
             
            <span
              className="highlight"
            >
               På afveje 
            </span>
            med forskellige kendte personer, og har også deltaget i flere selvhjælpsserier. Skrev i 2005 bogen
            <span
              className="highlight"
            >
               Et liv på kanten 
            </span>
          </p>
        </div>
      </div>
      <div
        className="col-xs-12 celeb-bottom"
      >
        <CarouselSlider
          onNextBook={[Function]}
          slideIndex={null}
        />
      </div>
    </div>
    <div
      className="col-xs-8 bookswrap"
    >
      <Pulse
        active=""
        onClick={[Function]}
        pid="870970-basis:52038014"
        position={
          Object {
            "x": 26,
            "y": 46,
          }
        }
      />
      <Pulse
        active=""
        onClick={[Function]}
        pid="870970-basis:23211629"
        position={
          Object {
            "x": 36.5,
            "y": 46,
          }
        }
      />
    </div>
  </div>
</section>
`;
